apps/rule-manager/app/service/RuleManager.scala

```diff
@@ -349,4 +355,51 @@ object RuleManager extends Loggable {
       case e: Exception => Left(e)
     }
   }
+
+  def destructivelyPublishDictionaryRules(
+      words: List[String],
+      bucketRuleResource: BucketRuleResource
+  ) = {
+    // Destroy existing draft dictionary rules
+    val ids = DbRuleDraft.findAllDictionaryRules().map(rule => rule.id)
```

Comment:
I think it's worth writing a method for doing this; we're loading 250k rules into memory here, and then making 250k DB calls, which is ... lots.

We could also make the relation between the tag and the rule ON CASCADE DELETE to make this unnecessary – I think that's reasonable for a join table, where removing the rule will always entail removing the join (and removing rules is very rare!)

Reply:
Sounds wise, let's do that. To clarify, do you mean between the rule and the rule-tag?

Reply:
Nice, yep, ON CASCADE DELETE for the rule FK in a rule tag.

Reply:
That'd be fair, could be a follow up!